src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -833,6 +839,39 @@ private static ulong CurlSendCallback(IntPtr buffer, ulong size, ulong nitems, I
                 return Interop.Http.CURL_READFUNC_ABORT;
             }
 
+            private static void SetLowSpeedLimit(EasyRequest easy, TimeSpan timeout)
+            {
+                // If there's no send/receive timeout on the handler, the defaults will
+                // remain disabled, and we don't need to worry about setting or resetting.
+                if (!easy._handler._hasSendReceiveTimeout)
+                {
+                    return;
+                }
+
+                // We need to ensure the timeout is set appropriately, which may include
+                // disabling a timeout currently set if one was.
+
+                long limit = 0, time = 0; // 0 == disable
+
+                if (timeout != Timeout.InfiniteTimeSpan)
+                {
+                    double secs = timeout.TotalSeconds;
+                    if (secs >= 1)
+                    {
+                        limit = 1;
```

Comment:
> Based on this conversation, we should probably pause and re-consider some of this.

Ok, I'm going to close this PR.

> Clearly, we need to make sure that "timeouts" (CancellationToken) on the stream.Read* of the response body and/or request body work as intended. I think that currently functionality is broken right now.

That works right now on Unix.  You're right that the token isn't respected on Windows.

For the outlined scenarios, it seems like just respecting the CancellationTokens is sufficient (in addition to exposing ConnectTimeout, which also seems reasonable if we believe that's an issue folks actually face).  Adding implementations of Stream.Read/WriteTiimeout would then logically just be a nicety on top of that (e.g. if you wanted to timeout individual read/write operations that made up a CopyToAsync without reimplementing CopyToAsync), and we could potentially implement them more efficiently internally than externally.